fastlane/lib/fastlane/actions/github_api.rb

```diff
@@ -0,0 +1,266 @@
+module Fastlane
+  module Actions
+    module SharedValues
+      GITHUB_API_STATUS_CODE = :GITHUB_API_STATUS_CODE
+      GITHUB_API_RESPONSE = :GITHUB_API_RESPONSE
+      GITHUB_API_JSON = :GITHUB_API_JSON
+    end
+
+    class GithubApiAction < Action
+      def self.run(params)
+        require 'json'
+
+        http_method = (params[:http_method] || 'GET').to_s.upcase
+        url = parse_url(params[:server_url], params[:path], params[:url])
+        headers = parse_headers(params[:api_token], params[:headers])
+        request_body = parse_body(params[:body], params[:raw_body])
+        handled_errors = params[:errors] || {}
+
+        response = call_endpoint(
+          url,
+          http_method,
+          headers,
+          request_body,
+          { secure: params[:secure], debug: params[:debug] }
+        )
+
+        status_code = response[:status]
+        result = {
+          status: status_code,
+          body: response.body || "",
+          json: parse_json(response.body) || {}
+        }
+
+        if status_code.between?(200, 299)
+          if params[:debug]
```

Comment:
ah - thanks, I forgot that was auto-logged on debug mode - updated. I wasn't aware of the global for verbose, so yup that will work perfectly in this case too (https://github.com/fastlane/fastlane/pull/8971#discussion_r122054377). I've removed the debug parameter for two of the actions and will now use this to discover when to print request/response bodies, much cleaner.